File: back/Canvas/Modules/Canvas/Interfaces.cs
﻿using Canvas.Controllers;

namespace Canvas.Modules.Canvas
{
    public interface ICanvasRepository
    {
        string GetUserCanvasList(UserData data);
        string GetCanvasById(CanvasByIdData data);
    }
    public interface ICanvasService
    {
        string CreateCanvas(CreateCanvasData data);
        string DeleteCanvas(DeleteCanvasData data);
        string UpdateCanvas(SaveCanvasData data);
    }
}

File: front/src/modules/canvas/interfaces.ts
export interface ICanvasRepository {
	getCanvasById(ownerId: string, id: string): Promise<ICanvasData>;
}

export interface ICanvasService {
		deleteCanvas(ownerId: string, canvasId: string): Promise<any>;
		createCanvas(ownerId: string, title: string, type: string): Promise<any>;
		setCanvasListAfterRemoving(canvasId: string): void;
		updateCanvas(canvasData: Record<string, any>): Promise<any>;
		downloadPdf(canvasData: Record<string, any>): Promise<any>;
}

export interface ICanvasData {
	id: string
	title: string;
	type: string;
	columns: number;
	rows: number;
	data: ICanvasBlocksData[];
}

export interface ICreateCanvasTemplate {
	ownerId: string | null;
	title: string | null;
	type: string | null;
	date: string | null;
	rows: number | null;
	columns: number | null;
	data: ICanvasBlocksData[];
}

export interface ICanvasBlocksData {
	position: number[];
	title: string;
	content: string;
	description: string;
}